db/main.go

```diff
@@ -0,0 +1,191 @@
+	}
+	if *goodleDriveOAuthTokenFile == "" {
+		log.Fatal("informe o path para o arquivo de token OAuth do Google Drive")
+	}
+	// Creating datastore client
+	datastoreClient, err := datastore.NewClient(context.Background(), *projectID)
+	if err != nil {
+		log.Fatalf("falha ao criar cliente de datastore: %v", err)
+	}
+	// Creating Google Drive client
+	googleDriveService, err := createGoogleDriveClient(*googleDriveCredentialsFile, *goodleDriveOAuthTokenFile)
+	if err != nil {
+		log.Fatalf("falha ao criar cliente do Google Drive, erro %q", err)
+	}
+	if err := summarize(*source, *state, datastoreClient, googleDriveService); err != nil {
+		log.Fatalf("falha ao executar processamento do resumidor do banco, erro %q", err)
+	}
+}
+
+func createGoogleDriveClient(googleDriveCredentialsFile, goodleDriveOAuthTokenFile string) (*drive.Service, error) {
+	b, err := ioutil.ReadFile(googleDriveCredentialsFile)
+	if err != nil {
+		log.Fatalf("falha ao ler arquivo de crendenciais [%s], erro %q", googleDriveCredentialsFile, err)
+	}
+	config, err := google.ConfigFromJSON(b, drive.DriveScope)
+	if err != nil {
+		return nil, fmt.Errorf("falha ao processar configuraçōes usando o arquivo [%s], erro %q", googleDriveCredentialsFile, err)
+	}
+	f, err := os.Open(goodleDriveOAuthTokenFile)
+	if err != nil {
+		return nil, fmt.Errorf("falha ao abrir arquivo de token oauth [%s], erro %q", goodleDriveOAuthTokenFile, err)
+	}
+	defer f.Close()
+	tok := &oauth2.Token{}
+	if err = json.NewDecoder(f).Decode(tok); err != nil {
+		return nil, fmt.Errorf("falha ao fazer bind do token OAuth, erro %q", err)
+	}
+	googleDriveService, err := drive.New(config.Client(context.Background(), tok))
+	if err != nil {
+		return nil, fmt.Errorf("não foi possível criar Google Drive service, erro %q", err)
+	}
+	return googleDriveService, nil
+}
+
+func summarize(source, state string, datastoreClient *datastore.Client, googleDriveService *drive.Service) error {
+	query := fmt.Sprintf("name contains '%s' and '%s' in parents", state, source) // pegando os arquivos com prefixo 'estado' da pasta de id 'source'
+	fileList, err := googleDriveService.Files.List().Q(query).Do()
+	if err != nil {
+		return fmt.Errorf("falha ao buscar arquivos do estado [%s] no diretório [%s], erro %q", state, source, err)
+	}
+	candFiles := getCandidateFiles(fileList)
+	dbItems, err := getDBItems(candFiles, googleDriveService)
+	if err != nil {
+		return fmt.Errorf("falha ao gerar itens do banco, erro %q", err)
+	}
+	for _, c := range dbItems {
+		userKey := datastore.NameKey(candidaturesCollection, fmt.Sprintf("%s_%s", c.State, c.City), nil)
+		if _, err := datastoreClient.Put(context.Background(), userKey, c); err != nil {
+			return fmt.Errorf("falha ao salvar cidade [%s] do estado [%s] no banco, erro %q", c.City, c.State, err)
+		}
+		log.Printf("saved city [%s] of state [%s]", c.City, c.State)
+	}
+	return nil
+}
+
+func getCandidateFiles(fileList *drive.FileList) map[string]*gDriveCandFiles {
```

Comment:
O retorno pode ser map[string]gDriveCandFiles .. isso vai tornar o  código abaixo ainda mais claro

```suggestion
func getCandidateFiles(fileList *drive.FileList) map[string]gDriveCandFiles {
```